exercises/npm-registry/src/package.ts

```diff
@@ -1,22 +1,53 @@
-import { RequestHandler } from 'express';
-import got from 'got';
-import { NPMPackage } from './types';
+import { RequestHandler } from "express";
+import got from "got";
+import { minSatisfying } from "semver";
+import { NPMPackage } from "./types";
 
 /**
  * Attempts to retrieve package data from the npm registry and return it
  */
 export const getPackage: RequestHandler = async function (req, res, next) {
   const { name, version } = req.params;
-
+  const dependencyTree = {};
   try {
     const npmPackage: NPMPackage = await got(
-      `https://registry.npmjs.org/${name}`,
+      `https://registry.npmjs.org/${name}`
```

Comment:
we can add the provided version to the request.
multiple reasons for that:
 - getting a specific version can be faster. 
 - less memory is consumed since getting all the versions can have a very large response.
 - in case package/version couldn't be found the api returns 404, which is the expected result in this case.
 
In this case we will fetch only the provided version we will have to create a new interface VersionedNPMPackage (without versions attribute, and dependencies on the root:
 `interface VersionedNPMPackage {
  name: string;
  description: string;
  'dist-tags': {
    [tag: string]: string;
  };
  dependencies?: NPMPackageDependencies;
}`) 

`type NPMPackageDependencies = Record<string, string>`


Also, the url should not be hard coded in the function, it should be configurable.
- Will be better for testing (we can provide a different url for testing, production)
- Same url can be used for getPackage/getDependencies
- in case the url changes we can change it one location and we won't need to deploy a new version of the server, just reset.